tests/hil/tests/settings/test_settings.py

```diff
@@ -207,3 +208,24 @@ async def test_cancel_all(board, device):
 
     await device.settings.set('TEST_INT', 72)
     assert None != board.wait_for_regex_in_line('Received test_int: 72', timeout_s=10)
+
+
+async def test_restart(board, device):
+    await device.settings.set('TEST_RESTART', True)
+    assert None != board.wait_for_regex_in_line('Received test_restart: true', timeout_s=10)
+    assert None != board.wait_for_regex_in_line('Ending session', timeout_s=10)
+
+    # Check that we no longer receive this settings change
+    await device.settings.set('TEST_INT', 2320)
+    with pytest.raises(RuntimeError) as e:
+        assert None != board.wait_for_regex_in_line('Received test_int', timeout_s=10)
```

Comment:
We can probably drop this timeout to 5 and reduce the 15 second timeout in the firmware to 10, if we're concerned about the length of the test. But I'm also fine with the current values.

Reply:
I'm also going to change this to test for the number that was set:

```suggestion
        assert None != board.wait_for_regex_in_line('Received test_int: 2320', timeout_s=10)
```